src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -96,6 +98,80 @@ int distance(BoundDecisionDagNode x)
             return result.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Enumerate the paths from the root node to the node of interest, and invokes the handler
+        /// on each one until the handler returns false.
+        /// The order is deterministic, but we're not starting from the shortest path.
+        /// </summary>
+        /// <param name="nodes">The set of nodes in topological order.</param>
+        /// <param name="targetNode">The node of interest.</param>
+        /// <param name="nullPaths">Whether to permit following paths that test for null.</param>
+        /// <param name="handler">Handler to call back for every pather to the target node.</param>
+        private static void VisitPathsToNode(ImmutableArray<BoundDecisionDagNode> nodes, BoundDecisionDagNode targetNode, bool nullPaths,
+            Func<ImmutableArray<BoundDecisionDagNode>, bool, bool> handler)
+        {
+            var pathBuilder = new ArrayBuilder<BoundDecisionDagNode>();
+            exploreToNode(nodes[0], currentRequiresFalseWhenClause: false);
+            return;
+
+            // Recursive exploration helper
+            // Returns true to continue, false to stop
+            bool exploreToNode(BoundDecisionDagNode currentNode, bool currentRequiresFalseWhenClause)
+            {
+                if (currentNode == targetNode)
+                {
+                    if (!handler(pathBuilder.ToImmutable(), currentRequiresFalseWhenClause))
```

Comment:
>if (!handler(pathBuilder.ToImmutable(), currentRequiresFalseWhenClause)) [](http://example.com/codeflow?start=20&length=72)

It feels like we should return `true` otherwise.

Reply:
The `BoundLeafDecisionDagNode` case below also handles all leaves.

Reply:
Replace with ```return handler(pathBuilder.ToImmutable(), currentRequiresFalseWhenClause);```?

Reply:
> The BoundLeafDecisionDagNode case below also handles all leaves.

But we reached the target, what is the point to continue? Can we reach it again?